lib/shopify_import_massager.ex

```diff
@@ -60,6 +60,120 @@ defmodule ShopifyImportMassager do
     end)
   end
 
+  def massage_all(
+        input_folder,
+        output_folder,
+        orders_files_filename_parameters = %{},
+        returns_files_filename_parameters = %{},
+        transactions_files_parameters = %{},
+        options \\ []
+      ) do
+    # We want the unchanged list of files so we can extract our substitutions
+    orders_input_files =
+      orders_files_filename_parameters
+      |> Map.merge(%{
+        input_path_prefix: input_folder,
+        output_path_prefix: "IGNORED",
+        number_padding_characters: 0
+      })
+      |> MassagerFilenameGenerator.generate_filename_pairs()
+      |> Enum.map(fn {input_filename, _output_filename} -> input_filename end)
+
+    substitution_map =
+      orders_input_files
+      |> Enum.map(fn file_name ->
+        extract_substitution_map(file_name)
+      end)
+      |> List.flatten()
+
+    show_substitutions(substitution_map, options)
+
+    IO.puts("Processing order files with parameters #{inspect(orders_files_filename_parameters)}")
+
+    %{
+      substitution_map: substitution_map,
+      input_folder: input_folder,
+      output_folder: output_folder,
+      filename_parameters: orders_files_filename_parameters
+    }
+    |> substitute()
+
+    IO.puts(
+      "Processing returns files with parameters #{inspect(returns_files_filename_parameters)}"
+    )
+
+    %{
+      substitution_map: substitution_map,
+      input_folder: input_folder,
+      output_folder: output_folder,
+      filename_parameters: returns_files_filename_parameters
+    }
+    |> substitute()
+
+    IO.puts(
+      "Processing returns files with parameters #{inspect(transactions_files_filename_parameters)}"
+    )
+
+    %{
+      substitution_map: substitution_map,
+      input_folder: input_folder,
+      output_folder: output_folder,
+      filename_parameters: transactions_files_filename_parameters
+    }
+    |> substitute()
+  end
+
+  def substitute(%{
```

Comment:
We're breaking up that long function functionality into smaller methods. If this needed to be a robust part of core code (instead of being used for one-off import data adjustments) then we would probably put this into modules of just a few functions and have unit tests for the very specific functionality.